src/app/content/components/ContentExcerpt.tsx

```diff
@@ -1,26 +1,47 @@
+import flow from 'lodash/fp/flow';
 import React from 'react';
+import { useSelector } from 'react-redux';
 import styled from 'styled-components/macro';
 import { bodyCopyRegularStyle } from '../../components/Typography';
-import addTargetBlankToLinks from '../utils/addTargetBlankToLinks';
+import { assertDefined } from '../../utils';
+import { book } from '../selectors';
+import { findArchiveTreeNodeById } from '../utils/archiveTreeUtils';
+import { addTargetBlankToLinks, fixRelativeURLs } from '../utils/contentManipulation';
+import { getBookPageUrlAndParams } from '../utils/urlUtils';
 
 interface Props {
   content: string;
   className: string;
+  sourcePageId: string;
 }
 
 // tslint:disable-next-line:variable-name
 const ContentExcerpt = styled((props: Props) => {
   const {
     content,
     className,
+    sourcePageId,
     ...excerptProps
   } = props;
 
-  const fixedContent = React.useMemo(
-    () => addTargetBlankToLinks(props.content),
-    [props.content]
+  const currentBook = useSelector(book);
+
+  if (!currentBook) {
```

Comment:
is there a legitimate situation where book would not be loaded? if not then we can throw here